### PR TITLE
fix code bug 22

### PR DESCRIPTION
**Pull request by zhaowei1936**

### CODE REVIEW

**Comment by zhaowei1936** [comment]
[ $ERROR_COUNT -gt 0 ]   && echo -e "  ${RED}✗ ERRORS:   $ERROR_COUNT${NC}  (必须修复才能编译)"
[ $WARNING_COUNT -gt 0 ] && echo -e "  ${YELLOW}⚠ WARNINGS: $WARNING_COUNT${NC}  (建议修复)"
echo -e "  ${GREEN}✓ PASSED:   $PASS_COUNT${NC}"

**Comment by zhaowei1936** [comment]
https://www.hiascend.com/document/detail/zh/Pytorch/60RC1/configandinstg/instg/insg_0006.html

**Comment by zhaowei1936** [comment]
conda create -n xyh_py310 python=3.10

**Comment by zhaowei1936** [comment]
具体而言：

* 当 Dot = 0 时，阶码不占据位宽，且 E 直接等于 0；
* 当 Dot = 1\~4 时，固定阶码幅值的最高位为 1（图 1 中的红色数字）。

在此规则下：

* 当 Dot = 1 时，阶码 E 只能表示 ±1，不包含已经编码的 0；
* 当 Dot = 2 时，阶码 E 只能表示 ±[2, 3]，不包含已经编码的 [-1, 1]。以此类推，Dot 域指示的 0\~4 共五种位宽阶码，共编码了 [-15,  15] 的指数，且彼此之间不重复，实现了无冗余编码。

**Comment by zhaowei1936** [comment]
ps -ef | grep alltoall_test | grep -v grep | awk '{print $2}' | xargs kill -9
ps -ef | grep python | grep -v grep | awk '{print $2}' | xargs kill -9
ps -ef | grep VLLM* | grep -v grep | awk '{print $2}' | xargs kill -9

**Comment by zhaowei1936** [comment]
def deepseek_v4_pro_debug_16_layers() -> TrainerConfig:
    return TrainerConfig(
        hf_assets_path="/mnt/share/Deepseek-v4-pro-tokenizer",
        model_spec=model_registry("v4_pro_debug_16_layers"),
        model_converters=ModelConvertersContainer.Config(
            converters=_default_converters()
        ),
        metrics=MetricsProcessor.Config(log_freq=1),
        dataloader=HuggingFaceTextDataLoader.Config(dataset="c4_test"),
        optimizer=OptimizerConfig(
            name="AdamW",
            lr=1e-5,
            eps=1e-6,
            swap_optimizer=True,
        ),
        lr_scheduler=LRSchedulersContainer.Config(
            warmup_steps=4,
            decay_ratio=0.8,
            decay_type="cosine",
            min_lr_factor=0.01,
        ),
        training=TrainingConfig(
            local_batch_size=1,
            seq_len=4096,
            max_norm=1.0,
            steps=20,
            num_mtp_modules=1,
        ),
        parallelism=ParallelismConfig(
            data_parallel_replicate_degree=1,
            data_parallel_shard_degree=-1,
            fsdp_reshard_after_forward="always",
            tensor_parallel_degree=1,
            enable_async_tensor_parallel=False,
            pipeline_parallel_degree=1,
            pipeline_parallel_schedule="Interleaved1F1B",
            expert_parallel_degree=16,
            expert_tensor_parallel_degree=1,
            context_parallel_degree=1,
        ),
        checkpoint=CheckpointManager.Config(
            enable=False,
            folder="checkpoint",
            load_step=0,
            interval=500,
            last_save_model_only=True,
            load_only=True,
            initial_load_in_hf=False,
            initial_load_path="/data/models/deepseek-v4-pro-bfloat16",
            export_dtype="float32",
            async_mode="disabled",
        ),
        activation_checkpoint=ActivationCheckpointConfig(mode="full"),
        compile=CompileConfig(enable=False, components=["model", "loss"]),
        profiling=ProfilingConfig(
            enable_profiling=False,
            enable_online_parse=False,
            profile_ranks=[0],
            profile_step_start=6,
            profile_step_end=7,
            profile_record_shapes=True,
            profile_with_memory=True,
            enable_memory_snapshot=False,
            save_memory_snapshot_folder="memory_snapshot",
        ),
    )

def deepseek_v4_pro_debug_61_layers_4k_384die() -> TrainerConfig:
    return TrainerConfig(
        hf_assets_path="./tests/assets/tokenizer/deepseek_v4_pro_tokenizer",
        model_spec=model_registry("v4_pro_debug_61_layers"),
        model_converters=ModelConvertersContainer.Config(
            converters=_default_converters()
        ),
        metrics=MetricsProcessor.Config(log_freq=1),
        dataloader=HuggingFaceTextDataLoader.Config(dataset="c4_test"),
        optimizer=OptimizerConfig(
            name="AdamW",
            lr=1e-5,
            eps=1e-6,
            swap_optimizer=True,
        ),
        lr_scheduler=LRSchedulersContainer.Config(
            warmup_steps=400,
            decay_ratio=0.8,
            decay_type="cosine",
            min_lr_factor=0.01,
        ),
        training=TrainingConfig(
            global_batch_size=384,
            local_batch_size=1,
            seq_len=4096,
            max_norm=1.0,
            steps=2000,
            num_mtp_modules=1,
        ),
        parallelism=ParallelismConfig(
            data_parallel_replicate_degree=1,
            data_parallel_shard_degree=-1,
            fsdp_reshard_after_forward="always",
            tensor_parallel_degree=1,
            enable_async_tensor_parallel=False,
            pipeline_parallel_degree=1,
            pipeline_parallel_schedule="Interleaved1F1B",
            expert_parallel_degree=384,
            expert_tensor_parallel_degree=1,
            context_parallel_degree=1,
        ),
        checkpoint=CheckpointManager.Config(
            enable=False,
            folder="checkpoint",
            load_step=0,
            initial_load_in_hf=True,
            initial_load_path="/data/models/deepseek-v4-pro-bfloat16",
            interval=10000,
            last_save_model_only=True,
            load_only=True,
            export_dtype="float32",
            async_mode="disabled",
        ),
        activation_checkpoint=ActivationCheckpointConfig(mode="full"),
        compile=CompileConfig(enable=True, components=["model", "loss"]),
        profiling=ProfilingConfig(
            enable_profiling=False,
            enable_online_parse=False,
            profile_ranks=[0],
            profile_step_start=6,
            profile_step_end=7,
            profile_record_shapes=True,
            profile_with_memory=True,
            enable_memory_snapshot=False,
            save_memory_snapshot_folder="memory_snapshot",
        ),
    )

**Comment by zhaowei1936** [comment]
{
  "OPENAI_API_KEY": "sk-b1475956f134e44c47d947b0af58f6c901a27dc1bcf0f248b79663ed7d06236c"
}

**Comment by zhaowei1936** [comment]
model_provider = "OpenAI"
model = "gpt-5.5"
review_model = "gpt-5.5"
model_reasoning_effort = "xhigh"
disable_response_storage = true
network_access = "enabled"
windows_wsl_setup_acknowledged = true

[model_providers.OpenAI]
name = "OpenAI"
base_url = "https://sub2api.aicorp.work"
wire_api = "responses"
requires_openai_auth = true

[features]
goals = true

**Comment by zhaowei1936** [comment]
%userprofile%\.codex/config.toml

**Comment by zhaowei1936** [comment]
%userprofile%\.codex/auth.json

**Comment by zhaowei1936** [comment]
按 Win+R，输入 %userprofile%\.codex 打开配置目录。如目录不存在，请先手动创建。
